config/v1/types_infrastructure.go

```diff
@@ -600,9 +600,300 @@ type OvirtPlatformStatus struct {
+	// Currently, we only support a single interface per RHCOS virtual machine.
+	// The available networks (port groups) can be listed using
+	// govc ls 'network/*'
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:MinItems=1
+	Networks []string `json:"networks,omitempty"`
+
+	// datastore is the name or inventory path of the datastore in which the
+	// virtual machine is created/located.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	Datastore string `json:"datastore,omitempty"`
+}
+
+type VSpherePlatformFailureDomainHosts struct {
+	// vmGroupName is the Virtual Machine Group name configured
+	// within a vCenter cluster that is associated with
+	// the corresponding Host Group.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	VMGroupName string `json:"vmGroupName"`
+
+	// hostGroupName is the Host Group name configured
+	// within a vCenter cluster defining a group
+	// of ESXi hosts.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	HostGroupName string `json:"hostGroupName"`
+}
+
+// VSpherePlatformVCenterSpec stores the vCenter connection fields.
+// This is used by the vSphere CCM.
+type VSpherePlatformVCenterSpec struct {
+
+	// server is the fully-qualified domain name or the IP address of the vCenter server.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=255
+	Server string `json:"server"`
+
+	// port is the TCP port that will be used to communicate to
+	// the vCenter endpoint.
+	// When omitted, this means the user has no opinion and
+	// it is up to the platform to choose a sensible default,
+	// which is subject to change over time.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=32767
+	// +optional
+	Port int `json:"port,omitempty"`
+
+	// The vCenter Datacenters in which the RHCOS
+	// vm guests are located. This field will
+	// be used by the Cloud Controller Manager.
+	// Each datacenter listed here should be used within
+	// a topology.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Datacenters []string `json:"datacenters"`
+}
+
+// VSpherePlatformPlacementConstraint is the context information for VM placements within a failure domain
+type VSpherePlatformPlacementConstraint struct {
+	// resourcePool is the absolute path of the resource pool where virtual machines will be
+	// created. The absolute path is of the form /<datacenter>/host/<cluster>/Resources/<resourcepool>.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:Pattern=`^/.*?/host/.*?/Resources.*`
+	// +optional
+	ResourcePool string `json:"resourcePool,omitempty"`
```

Comment:
So this was something that is not configurable before and now it is.  Can the values here conflict with folder optional below? If unspecified - the system picks a default ?

Reply:
ResourcePool is configurable
https://github.com/openshift/installer/blob/release-4.11/pkg/types/vsphere/platform.go#L42-L44